app/models/user_action_event.rb

```diff
@@ -6,4 +6,21 @@ class UserActionEvent < ApplicationRecord
   validates :details, presence: true
   validates :identifier, presence: true, uniqueness: true
   validates :event_type, presence: true
+
+  def self.setup
+    config_file_path = Rails.root.join('config', 'user_action_events.yml')
+    raise 'Config file not found' unless File.exist?(config_file_path)
+
+    user_action_events_yaml = YAML.load_file(config_file_path)
+
+    user_action_events_yaml.each do |identifier, event_config|
+      event = UserActionEvent.find_or_initialize_by(identifier:)
+      if event.new_record? || event.attributes.slice(*event_config.keys.map(&:to_s)) != event_config
```

Comment:
I think you can do this this way a bit simpler:
`event = UserActionEvent...`
`event.attributes = event_config`
`event.save! if event.changed?`